File: mergeSort/mergeSort.py
import random

def mergeSort(lista):
    if len(lista) > 1:
        medio = len(lista)//2
        derecha = lista[medio:]
        izquierda = lista[:medio]

        #llamado recursivo
        mergeSort(derecha);
        mergeSort(izquierda);

        #iteradores
        i = 0 #iterador para derecha
        j = 0 #iterador para izquierda

        k = 0 #iterador lista principal

        while i < len(derecha) and j < len(izquierda):
            if derecha[i] < izquierda[j]:
                lista[k] = derecha[i]
                i+=1
            else:
                lista[k] = izquierda[j]
                j+=1
            k+=1
        while i < len(derecha):
            lista[k] = derecha[i]
            i+=1
            k+=1
        while j < len(izquierda):
            lista[k] = izquierda[j]
            j+=1
            k+=1
    return lista


if __name__ == "__main__":
    tamano_de_lista = int(input('De que tamano sera la lista?'))

    lista = [random.randint(0,100) for i in range(tamano_de_lista)]
    print(lista)
    print('-' * 20)

    lista_ordenada = mergeSort(lista)
    print(lista_ordenada)

File: mergeSort/mergeSort.js
const mergeSort = (lista) => {
  if (lista.length > 1) {
    let half = Math.floor(lista.length / 2);
    let izquierda = lista.slice(0, half);
    let derecha = lista.slice(half);

    // Recursion
    mergeSort(izquierda);
    mergeSort(derecha);

    // indexing for sublists
    let i = 0;
    let j = 0;
    // indexing for main list
    let k = 0;

    while (i < izquierda.length && j < derecha.length) {
      if (derecha[j] < izquierda[i]) {
        lista[k] = derecha[j];
        j++;
      } else {
        lista[k] = izquierda[i];
        i++;
      }
      k++;
    }
    while (i < izquierda.length) {
      lista[k] = izquierda[i];
      i++;
      k++;
    }
    while (j < derecha.length) {
      lista[k] = derecha[j];
      j++;
      k++;
    }
  }
  return lista;
};

const main = () => {
  let array = [];
  for (let i = 0; i < 10; i++) {
    const random = Math.floor(Math.random() * 100);
    array.push(random);
  }
  console.log(array);
  console.log("-----------------------------------");
  console.log(mergeSort(array));
};

main();
